includes/specials/SpecialUndelete.php

```diff
@@ -1452,6 +1453,9 @@ function undelete() {
 			$out->addWikiMsg( 'undeleterevdel' );
 		}
 
+		$redirKey = wfMemcKey( 'redir', $this->mTargetObj->getPrefixedText() );
+		$wgMemc->delete( $redirKey );
```

Comment:
Using the `UndeleteComplete` hook to do this would be preferred over modifying the core MW file and having this code know about an extension-specific Memcache key.